keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/ImageService.kt

```diff
@@ -52,10 +52,10 @@ class ImageService(
    *
    * If no images at all exist for [artifact] this method returns `null`.
    */
-  suspend fun getLatestImage(artifact: DeliveryArtifact, account: String): Image? {
+  suspend fun getLatestImage(artifact: DebianArtifact, account: String): Image? {
     val candidateImages = cloudDriverService
       .namedImages(DEFAULT_SERVICE_ACCOUNT, artifact.name, account)
-      .filter { it.hasAppVersion && it.hasBaseImageName }
+      .filter { it.hasAppVersion && it.hasBaseImageName && it.baseImageName.startsWith("${artifact.vmOptions.baseOs}base") }
```

Comment:
filter so that we only consider images with the same base OS

Reply:
I specifically made it a substring with `base` because `bionic` would match `bionic-classicbase` otherwise